dask/dataframe/accessor.py

```diff
@@ -135,15 +135,23 @@ def split(self, pat=None, n=-1, expand=False):
 
     @derived_from(pd.core.strings.StringMethods)
     def cat(self, others=None, sep=None, na_rep=None):
-        from .core import Series, Index
+        from .core import DataFrame, Series, Index
 
         if others is None:
             raise NotImplementedError("x.str.cat() with `others == None`")
 
-        valid_types = (Series, Index, pd.Series, pd.Index)
-        if isinstance(others, valid_types):
+        is_dataframe_like = (DataFrame, pd.DataFrame)
+        is_series_like = (Series, pd.Series)
+        is_index_like = (Index, pd.Index)
+
+        if isinstance(others, is_series_like) or isinstance(others, is_index_like):
```

Comment:
Replace the isinstance checks with `is_series_like`, `is_index_like` from `dask.dataframe.utils`

Reply:
Thx for the reference - updated the file.  
maybe later we can try to implement `.str.cat` with a `DataFrame` - will need to check how Pandas implements it.

Reply:
What happens if you just pass `others` through to 

```
        return self._series.map_partitions(
            str_cat, *others, sep=sep, na_rep=na_rep, meta=self._series._meta
        )
```

even when it's a DataFrame?

Reply:
Hi,
When we tried to pass a `dask.dataframe` we get the following error:  


> ....
>~/projects/dask/dask/dataframe/accessor.py in str_cat(self, *others, **kwargs)
>     183 
>     184 def str_cat(self, *others, **kwargs):
> --> 185     return self.str.cat(others=others, **kwargs)
> 
> ~/anaconda3/envs/dask-dev/lib/python3.8/site-packages/pandas/core/strings.py in wrapper(self, *args, **kwargs)
>    1841                 )
>    1842                 raise TypeError(msg)
> -> 1843             return func(self, *args, **kwargs)
>    1844 
>    1845         wrapper.__name__ = func_name
> 
> ~/anaconda3/envs/dask-dev/lib/python3.8/site-packages/pandas/core/strings.py in cat(self, others, sep, na_rep, join)
>    2346         try:
>    2347             # turn anything in "others" into lists of Series
> -> 2348             others, warn = self._get_series_list(others, ignore_index=(join is None))
>    2349         except ValueError:  # do not catch TypeError raised by _get_series_list
>    2350             if join is None:
> 
> ~/anaconda3/envs/dask-dev/lib/python3.8/site-packages/pandas/core/strings.py in _get_series_list(self, others, ignore_index)
>    2166                     # because "x in df" returns column names
>    2167                     if not is_legal or isinstance(nxt, DataFrame):
> -> 2168                         raise TypeError(err_msg)
>    2169 
>    2170                     nxt, wnx = self._get_series_list(nxt, ignore_index=ignore_index)
> 
> TypeError: others must be Series, Index, DataFrame, np.ndarray or list-like (either containing only strings or containing only objects of type Series/Index/list-like/np.ndarray)

From what we debugged/understand in order for `str.cat` to accept a `dataframe` it currently *must* be a `pandas`  dataframe and not a `dask` (from both sides of the function).   

If you think  there is a solution - will be happy to improve the PR with some guidance.

Reply:
Passing a DataFrame through certainly seems doable, but I won't be able to look closely in the near-term.

Reply:
OK - do you want to accept the PR as-is or wait until the dataframe  is implemented ?

Reply:
I’d prefer to wait until it’s implemented.

Reply:
OK - if you know of a similar functionality that was developed we will be happy to review it and try implementing it in this PR